pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/inv/OffHeapBitmapInvertedIndexCreator.java

```diff
@@ -181,33 +184,45 @@ public void seal()
     }
 
     // Create bitmaps from inverted index buffers and serialize them to file
-    try (DataOutputStream offsetDataStream = new DataOutputStream(
-        new BufferedOutputStream(new FileOutputStream(_invertedIndexFile)));
-        FileOutputStream bitmapFileStream = new FileOutputStream(_invertedIndexFile);
-        DataOutputStream bitmapDataStream = new DataOutputStream(new BufferedOutputStream(bitmapFileStream))) {
-      int bitmapOffset = (_cardinality + 1) * Integer.BYTES;
-      offsetDataStream.writeInt(bitmapOffset);
-      bitmapFileStream.getChannel().position(bitmapOffset);
-
+    ByteBuffer offsetBuffer = null;
+    ByteBuffer bitmapBuffer = null;
+    try (FileChannel channel = new RandomAccessFile(_invertedIndexFile, "rw").getChannel()) {
+      // map the offsets buffer
+      int startOfBitmaps = (_cardinality + 1) * Integer.BYTES;
+      offsetBuffer = channel.map(FileChannel.MapMode.READ_WRITE, 0, startOfBitmaps)
+              .order(ByteOrder.BIG_ENDIAN);
+      offsetBuffer.putInt(startOfBitmaps);
+      bitmapBuffer = channel.map(FileChannel.MapMode.READ_WRITE, startOfBitmaps, Integer.MAX_VALUE - startOfBitmaps)
+              .order(ByteOrder.LITTLE_ENDIAN);
+      RoaringBitmapWriter<RoaringBitmap> writer = RoaringBitmapWriter.writer().get();
```

Comment:
I see your point. We won't create bitmap inverted index if the values are sorted (we have another RLE format), so in most cases, the docIds will spread over all the arrays. But for high cardinality case where each bitmap only contain very few values, pre-allocation might end up allocating too many arrays. So agree with letting it grow automatically.

Reply:
If that's the case, currently this will try to RLE each container when it's appended to the bitmap. Would you prefer I remove that when I reformat the file?

Reply:
I disabled run compression, which will save some cycles if this is only used for sparse bitmaps.

Reply:
The bitmap could be dense (when cardinality is very low), but won't be contiguous. What is the old behavior? We don't explicitly call `runOptimize()` on the `MutableRoaringBitmap`, will it perform RLE automatically? If not, we can disable it here to keep the same behavior and have better performance.

Reply:
No run length encoding was being done before